json.go

```diff
@@ -137,3 +137,11 @@ func SyntaxErrorOffset(err error) (isSyntaxError bool, offset int64) {
 		return false, 0
 	}
 }
+
+// FieldError are errors that provide access to
+// the path of the erroneous field
```

Comment:
```suggestion
// FieldError is an error that provides access to the path of the erroneous field
```

Reply:
update the godoc for UnmarshalStrict that returned strict errors about specific fields will implement the FieldError interface, and add type assertions to the unit tests in this package that that is the case

Reply:
done